src/main/java/de/tum/in/www1/artemis/web/rest/AchievementResource.java

```diff
@@ -0,0 +1,135 @@
+public class AchievementResource {
+
+    private static final String ENTITY_NAME = "achievement";
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    private final Logger log = LoggerFactory.getLogger(AchievementResource.class);
+
+    private final AuthorizationCheckService authCheckService;
+
+    private final AchievementService achievementService;
+
+    private final UserService userService;
+
+    private final AchievementRepository achievementRepository;
+
+    public AchievementResource(AuthorizationCheckService authCheckService, AchievementService achievementService, UserService userService,
+            AchievementRepository achievementRepository) {
+        this.authCheckService = authCheckService;
+        this.achievementService = achievementService;
+        this.userService = userService;
+        this.achievementRepository = achievementRepository;
+    }
+
+    /**
+     * GET /achievements : Get all achievements for user.
+     *
+     * @return the ResponseEntity with status 200 (OK) and with the found achievements as body
+     */
+    @GetMapping("/achievements")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Set<Achievement>> getAchievementsForUser() {
+        User user = userService.getUser();
+        log.debug("REST request to get achievements for user : {}", user.getLogin());
+        Set<Achievement> achievements = achievementService.findAllForUser(user.getId());
+        return ResponseEntity.ok(achievements);
+    }
+
+    /**
+     * GET /courses/{courseId}/achievements : Get all achievements for course.
+     *
+     * @param courseId the course to which the achievements belong to
+     * @return the ResponseEntity with status 200 (OK) and with the found achievements as body
+     */
+    @GetMapping("/courses/{courseId}/achievements")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Set<Achievement>> getAchievementsForCourse(@PathVariable Long courseId) {
+        log.debug("REST request to get achievements for course : {}", courseId);
+        Set<Achievement> achievements = achievementService.findAllForCourse(courseId);
+        return ResponseEntity.ok(achievements);
+    }
+
+    /**
+     * PUT /achievements : Updates an existing achievement.
+     *
+     * @param achievement the achievement to update
+     * @return the ResponseEntity with status 200 (OK) and with body the updated achievement, or with status 500
+     *         (Internal Server Error) if the achievement couldn't be updated
+     */
+    @PutMapping("/achievements")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Achievement> updateAchievement(@RequestBody Achievement achievement) {
+        log.debug("REST request to update Achievement : {}", achievement);
+
+        if (achievement.getId() == null) {
+            return ResponseEntity.badRequest().build();
+        }
```

Comment:
I would add an validation for the rank which is defined as INT at the moment. For example only allow ranks >0 or use the enum approach explained above.

Reply:
Went for the enum approach :)